MonoGame.Framework.Content.Pipeline/Builder/ContentBuilderHelper.cs

```diff
@@ -0,0 +1,289 @@
+
+    static ContentBuilderHelper()
+    {
+        var serilizer = new SerializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .WithTypeConverter(new ColorConverter())
+            .EnablePrivateConstructors()
+            .DisableAliases();
+
+        var deserializer = new DeserializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .WithTypeConverter(new ColorConverter())
+            .EnablePrivateConstructors()
+            .IgnoreFields()
+            .IgnoreUnmatchedProperties();
+
+        foreach (var a in AppDomain.CurrentDomain.GetAssemblies())
+        {
+            foreach (var t in a.GetTypes())
+            {
+                if (t.IsAbstract || t.IsInterface)
+                    continue;
+
+                if (t.GetInterface(nameof(IContentImporter)) != null)
+                {
+                    serilizer.WithTagMapping("!" + t.ToString(), t);
+                    deserializer.WithTagMapping("!" + t.ToString(), t);
+
+                    _importers.Add(new ImporterInfo
+                    {
+                        Attribute = GetImporterAttribute(t),
+                        Type = t
+                    });
+                }
+                else if (t.GetInterface(nameof(IContentProcessor)) != null)
+                {
+                    serilizer.WithTagMapping("!" + t.ToString(), t);
+                    deserializer.WithTagMapping("!" + t.ToString(), t);
+
+                    _processors.Add(new ProcessorInfo
+                    {
+                        Attribute = GetProcessorAttribute(t),
+                        Type = t
+                    });
+                }
+                else if (t.IsSubclassOf(typeof(ContentServer)))
+                {
+                    var props = new List<ServerPropertyInfo>();
+                    foreach (var propInfo in t.GetProperties())
+                    {
+                        if (!propInfo.CanRead || !propInfo.CanWrite)
+                        {
+                            continue;
+                        }
+
+                        var attributes = propInfo.GetCustomAttributes(typeof(ContentServerParameterAttribute), false);
+                        if (attributes.Length == 0)
+                        {
+                            continue;
+                        }
+
+                        props.Add(new ServerPropertyInfo
+                        {
+                            Attribute = (ContentServerParameterAttribute)attributes[0],
+                            PropertyInfo = propInfo
+                        });
+                    }
+                    _serverOptions[t] = props;
+                }
+            }
+        }
+
+        Serializer = serilizer.Build();
+        Deserializer = deserializer.Build();
+    }
+
+    public static readonly ISerializer Serializer;
+
+    public static readonly IDeserializer Deserializer;
```

Comment:
Does this mean the builder can only work single threaded?

Reply:
No? Not sure why that part would prevent it from working multithreaded.

As for compiling multiple content files at the same time, we should be able to set it up, the only real issue is proper logging and adding some logic to chose which files to go to which thread, which is really outside of the scope of what this is doing.

Reply:
I assumed that these need to be different per asset type and being globals they would need to change per-asset.

But i'm probably misunderstanding how these are used.